Fix translations and UpdatedVersionAvailableUserAlert.updateNowButton

ArneBab opened the pull request:
It used the FALLBACK_LANG to get the fallback lang, but
iterator().next() skipped to the next value, which is the raw key.

This had also broken our whole translation system.

vwoodzell commented:
I agree with the change to MainJarUpdater. I can't imagine why we'd want to show the fallback string there instead of the localized one.

I don't think that BaseL10n.getDefaultString() should be changed, though. It doesn't return the raw key unless the key is missing from the fallback translation. I've added [tests](https://github.com/vwoodzell/fred-staging/commit/a6f6e2e7a2ab88b39936e22b808da92e11e5362f) that show it working correctly.

ArneBab commented:
Thank you for checking! I merged your tests and reverted the change to getDefaultString(). Also I changed the return value documentation to match the implementation.

Good to go? :-)

vwoodzell reviewed (approved):
Haha, good idea on the documentation change.
